firebase.json

```diff
@@ -24,6 +24,81 @@
           "source": "/hardware/catalog/:board*",
           "destination": "/firmware/hardware/catalog/:board",
           "type": 301
+        },
+        {
+          "source": "/hardware{,/**}",
+          "destination": "/firmware/hardware",
+          "type": 301
+        },
+        {
+          "source": "/cloud/output-streams/:streams*",
+          "destination": "/data-routing/output-streams/:streams",
+          "type": 301
+        },
+        {
+          "source": "/cloud/output-streams{,/**}",
+          "destination": "/data-routing/output-streams/",
+          "type": 301
+        },
+        {
+          "source": "/cloud/services{,/**}",
+          "destination": "/data-routing/application-services/",
+          "type": 301
+        },
+        {
+          "source": "/cloud/services/lightdb/:state*",
+          "destination": "/data-routing/application-services/lightdb-state/:state",
+          "type": 301
+        },
+        {
+          "source": "/services/lightdb{,/**}",
+          "destination": "/data-routing/application-services/lightdb-state/",
+          "type": 301
+        },
+        {
+          "source": "/cloud/services/lightdb-stream/:stream*",
+          "destination": "/data-routing/application-services/lightdb-stream/:stream",
+          "type": 301
+        },
+        {
+          "source": "/cloud/services/lightdb-stream{,/**}",
+          "destination": "/data-routing/application-services/lightdb-stream/",
+          "type": 301
+        },
+        {
+          "source": "/cloud/services/logging/:logs*",
+          "destination": "/device-management/logging/:logs",
+          "type": 301
+        },
+        {
+          "source": "/cloud/services/logging{,/**}",
+          "destination": "/device-management/logging",
+          "type": 301
+        },
+        {
+          "source": "/cloud/services/ota/:ota*",
+          "destination": "/device-management/ota/:ota",
+          "type": 301
+        },
+        {
+          "source": "/cloud/services/ota{,/**}",
+          "destination": "/device-management/ota",
+          "type": 301
+        },
+        {
+          "source": "/cloud/services/settings/:sets*",
+          "destination": "/device-management/settings/:sets",
+          "type": 301
+        },
+        {
+          "source": "/cloud/services/settings{,/**}",
+          "destination": "/device-management/settings",
+          "type": 301
+        },
+        {
+          "source": "/cloud{,/**}",
+          "destination": "/data-routing",
+          "type": 301
```

Comment:
@szczys are all of these rules processed in order (thus making this one a catch-all for cloud routes that do not match a previous rule)?

Reply:
Yes, that is correct. I thought it would be better to redirect pages to the root folder (if we miss an explicit redirect) rather than have them default to Page Not Found. Is that an okay assumption?

Reply:
I think so!